src/adagrams.js

```diff
@@ -1,15 +1,150 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let playersHand = [];
+  const letterPool = getLetterPool(LETTER_POOL);
+
+  while (playersHand.length !== 10) {
+    let letter = randomLetter(letterPool);
+    playersHand.push(letter);
+  }
+
+  return playersHand;
 };
 
+// grab numbers and then make sure we can use that letter
+
 export const usesAvailableLetters = (input, lettersInHand) => {
+  input = input.toUpperCase();
+  let availableBank = counter(lettersInHand);
+  for (const letter of input) {
+    if (letter in lettersInHand || availableBank[letter]) {
+      availableBank[letter] -= 1;
+    } else {
+      return false;
+    }
+  }
+  return true;
   // Implement this method for wave 2
 };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
-};
 
+  word = word.toUpperCase();
+
+  const ps = {};
+
+  ps.A = ps.E = ps.I = ps.O = ps.U = ps.L = ps.N = ps.R = ps.S = ps.T = 1;
+  ps.D = ps.G = 2;
+  ps.B = ps.C = ps.M = ps.P = 3;
+  ps.F = ps.H = ps.V = ps.W = ps.Y = 4;
+  ps.K = 5;
+  ps.J = ps.X = 8;
+  ps.Q = ps.Z = 10;
+  let score = 0;
+
+  for (const letter of word) {
+    for (const [key, point] of Object.entries(ps)) {
+      if (letter === key) {
+        score += point;
+      }
+    }
+  }
+  if (6 < word.length && word.length <= 10) {
+    score += 8;
+  }
+  return score;
+};
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  let scores = {};
+
+  for (const word of words) {
+    let key = scoreWord(word);
+    if (scores[key] === undefined) {
+      let list = [];
+      list.push(word);
+      scores[key] = list;
+    } else {
+      scores[key].push(word);
+    }
+  }
+  const highestScore = Math.max(...Object.keys(scores));
+
+  if (scores[highestScore].length > 1) {
+    let result;
+
+    for (const word of scores[highestScore]) {
+      if (word.length === 10) {
+        return { word: word, score: highestScore };
+      } else {
+        const minLen = Math.min(
+          ...scores[highestScore].map((word) => word.length)
+        );
+        if (word.length === minLen) {
+          result = { word: word, score: highestScore };
```

Comment:
Note that in JS objects, when your key name is the variable name of the value, you can use shorthand syntax:
```javascript
{ word, score: highestScore }
```